builder/azure/arm/config.go

```diff
@@ -1240,6 +1259,30 @@ func assertRequiredParametersSet(c *Config, errs *packersdk.MultiError) {
 	default:
 		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("The disk_caching_type %q is invalid", c.DiskCachingType))
 	}
+
+	/////////////////////////////////////////////
+	// License Type (Azure Hybrid Benefit)
+	if c.LicenseType != "" {
+		// Assumes OS is case-sensitive match as it has already been
+		// normalized earlier in function
+		if c.OSType == constants.Target_Linux {
+			if strings.EqualFold(c.LicenseType, constants.License_RHEL) {
+				c.LicenseType = constants.License_RHEL
+			} else if strings.EqualFold(c.LicenseType, constants.License_SUSE) {
+				c.LicenseType = constants.License_SUSE
+			} else {
+				errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("The license_type %q is invalid for os_type %q", c.LicenseType, c.OSType))
+			}
+		} else if c.OSType == constants.Target_Windows {
+			if strings.EqualFold(c.LicenseType, constants.License_Windows_Client) {
+				c.LicenseType = constants.License_Windows_Client
+			} else if strings.EqualFold(c.LicenseType, constants.License_Windows_Server) {
+				c.LicenseType = constants.License_Windows_Server
+			} else {
+				errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("The license_type %q is invalid for os_type %q", c.LicenseType, c.OSType))
```

Comment:
I think this error message can be a bit more specific

```suggestion
				errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("The license_type %q is invalid for Windows, use Windows_Client or Windows_Server", c.LicenseType))
```